benchmark/sprintf.rb

```diff
@@ -0,0 +1,43 @@
+require "benchmark/ips"
+require "lib/logstash/event"
+
+options = { :time => 10, :warmup => 10 }
+Benchmark.ips do |x|
+  x.config(options)
+  event = LogStash::Event.new("foo" => "bar",
+                    "foobar" => "morebar")
+
+  x.report("Complex No cache: Event#old_sprintf") { event.old_sprintf("/first/%{foo}/%{foobar}/%{+YYY-mm-dd}") }
+  x.report("Complex cached: Event#sprintf") { event.sprintf("/first/%{foo}/%{foobar}/%{+YYY-mm-dd}") }
+  x.compare!
+end
+
+Benchmark.ips do |x|
+  x.config(options)
+  event = LogStash::Event.new("foo" => "bar",
```

Comment:
I have updated the benchmark in the PR that create a new Event on each iteration.